### DISCUSSION TITLE
Obtaining an OAuth token

### DISCUSSION THREAD

**Discussion by hemenwaynick** [mozts2005/ZendeskApi_v2#513] [Features]
Is there a way to obtain an OAuth token through this library? Thanks

**Comment by mozts2005** [comment]
At this time no. But I see that Zendesk has added API endpoints for this. So it could be added.

https://developer.zendesk.com/rest_api/docs/support/oauth_tokens

We would except this as a community contribution￼.

**Comment by hemenwaynick** [comment]
Hi @mozts2005, I'm trying to write a test for some code I added for OAuth. Specifically, I'm exposing [Zendesk's password grant type API](https://support.zendesk.com/hc/en-us/articles/203663836#topic_z3q_shl_1l).
Do you have any thoughts on what I should use for the client secret? Should we avoid storing that in code? How about the client ID? Not sure what I should use for that field. The relevant lines of code are [here](https://github.com/hemenwaynick/ZendeskApi_v2/blob/efedaedff1733a4e23084962af92e7daa6f1092c/test/ZendeskApi_v2.Test/OAuthTokenTests.cs#L26).
Thanks

**Reply by mozts2005** [reply]
based on my reading of the full API

https://developer.zendesk.com/rest_api/docs/support/grant_type_tokens

I would ask that you make the request more generic to support creating any OAuth token. as you are not hard coding the  grant_type in the request. 

Next, given that the scopes are a limited list of valid values please use an enum not strings. 
https://developer.zendesk.com/rest_api/docs/support/grant_type_tokens#scope

Last,  I am not sure what you mean by "I should use for the client secret"
 Are you asking: 
- data type to store the value?
    String sound ok to me.

-  if we have client secret that you can use?
   No, we do not have any currently. Feel free to use the testing account to create one if needed.

- should the client secret needed for this test case be save to the setting class?
   Yes, just like all other testing data.

**Comment by hemenwaynick** [comment]
Do you have a client secret I can use?

**Reply by mozts2005** [reply]
No but you have full access to the test account. Feel free to create one.
